backends/transforms/test/test_create_mutable_buffer.py

```diff
@@ -0,0 +1,485 @@
+            buf_data: torch.Tensor,
+            return_buffer=False,
+        ):
+            super().__init__()
+            self.registered_buffers = set()
+            self.exported_program = exported_program
+            self.buf_data = buf_data
+            self.return_buffer = return_buffer
+
+        def call(self, graph_module: torch.fx.GraphModule):
+            """
+            This pass will register a mutable buffer for the add op(s) in the graph.
+            It will insert a new index_put_ op to the graph to update the buffer using the output of the add op.
+            And adjust the output of the graph to return the buffer or the index_put_ op.
+            """
+            modified = False
+            for node in graph_module.graph.nodes:
+                if (
+                    node.op == "call_function"
+                    and node.target == exir_ops.edge.aten.add.Tensor
+                ):
+                    # To match what we do in export.
+                    suffix = (
+                        ""
+                        if len(self.registered_buffers) == 0
+                        else f"{len(self.registered_buffers) + 1}"
+                    )
+                    buffer_name = f"b_my_buffer{suffix}"
+                    self.registered_buffers.add(buffer_name)
+
+                    # Utility under test!
+                    buf_node = create_mutable_buffer(
+                        self.exported_program,
+                        graph=graph_module.graph,
+                        data=self.buf_data,
+                        name=buffer_name,
+                    )
+
+                    # Assuming `indices` is always available
+                    indices = [
+                        node
+                        for node in graph_module.graph.nodes
+                        if node.op == "placeholder" and node.name == "indices"
+                    ][0]
+                    with graph_module.graph.inserting_after(node):
+                        index_put_node = graph_module.graph.create_node(
+                            "call_function",
+                            exir_ops.edge.aten.index_put_.default,
+                            (
+                                buf_node,
+                                (indices,),
+                                node,
+                            ),
+                            {},
+                        )
+                        # Update the output of the graph. This is a bit hacky, but it is done through `to_edge`.
+                        # We have to mimic it here for testing against eager.
+                        output_node = graph_module.graph.output_node()
+                        outputs = list(output_node.args[0])
+                        if self.return_buffer:
+                            outputs.insert(0, buf_node)
+                        else:  # return the index_put node
+                            outputs.insert(0, index_put_node)
```

Comment:
Yeah so why is this not done with create_mutable_buffer_node. Feels like this is an atomic thing that should be done together

Reply:
updated in the follow up.